plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/TableStatisticsMaker.java

```diff
@@ -94,7 +94,7 @@ private TableStatistics makeTableStatistics(IcebergTableHandle tableHandle, Cons
         for (int index = 0; index < partitionFields.size(); index++) {
             PartitionField field = partitionFields.get(index);
             Type type = icebergPartitionTypes.get(index);
-            idToDetailsBuilder.put(field.sourceId(), new ColumnFieldDetails(
+            idToDetailsBuilder.put(field.fieldId(), new ColumnFieldDetails(
```

Comment:
> Fix bug in Iceberg partition schema evolution


this will be squashed, right?

also, should we have a test, with two partitioning transformations over a column?

Reply:
Yeah, I'll put it in a separate PR though

Reply:
https://github.com/trinodb/trino/pull/10739